samples/go/ipfs-chat/pubsub.go

```diff
@@ -7,46 +7,55 @@ package main
 import (
 	"context"
 	"encoding/base64"
+	"strings"
 
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/datas"
 	"github.com/attic-labs/noms/go/hash"
-	"github.com/attic-labs/noms/go/ipfs"
 	"github.com/attic-labs/noms/go/merge"
 	"github.com/attic-labs/noms/go/types"
 	"github.com/attic-labs/noms/samples/go/ipfs-chat/dbg"
-	"gx/ipfs/QmddZj8gx1Ceisj3QEWeAzPEjCPWpXzneN1ANdkZdwUdc3/floodsub"
+	"github.com/ipfs/go-ipfs/core"
 )
 
-func Replicate(sub *floodsub.Subscription, source, dest datas.Dataset, didChange func(ds datas.Dataset)) {
-	lastHash := ""
+// MergeMessages continually listens for commit hashes published by ipfs-chat. It
+// merges new messages into it's existing dataset when necessary and if an actual
+// merge was necessary, it re-publishes the new commit.
+func MergeMessages(node *core.IpfsNode, topic string, ds datas.Dataset, didChange func(ds datas.Dataset)) {
```

Comment:
This function doesn't need to be public.

Reply:
Changed.